register/models.py

```diff
@@ -122,8 +139,14 @@ def send_reimbursement(self, request):
         self.save()
 
     def confirm(self, cancellation_url):
+        if self.status in [APP_ACCEPTED,APP_PENDING,APP_REJECTED]:
+            raise Http404('Unfortunately this invite doesn\'t even exist')
+        elif self.status == APP_CANCELLED:
+            raise ValidationError('This invite has been cancelled.')
+        elif self.status == APP_EXPIRED:
+            raise ValidationError('Unfortunately your invite has expired.')
         if self.status != APP_INVITED and self.status != APP_CONFIRMED:
-            raise ValidationError('Application hasn\'t been invited yet')
+            pass
         if self.status != APP_CONFIRMED:
             m = MailListManager()
             m.add_applicant_to_list(self, m.WINTER_17_LIST_ID)
```

Comment:
This might happen when status is APP_ATTENDED. I think it's better to change this condition to check `if self.status == APP_INVITED` or we'll have to touch it if we add new possible statuses.